src/gpu/intel/jit/emulation.hpp

```diff
@@ -591,13 +591,15 @@ struct EmulationImplementation {
         bool emulate64 = strategy.emulate64_mul;
 
         if (s0Q) {
-            if (!s1D || !dstQ) stub();
+            if (s1Q || !dstQ) stub();
             auto temp = s1Signed ? state.temp[0].d() : state.temp[0].ud();
             auto &src1Reg = [&]() -> ngen::RegData & {
-                if (std::is_base_of<ngen::RegData, S1>::value)
+                if (s1Immed || s1W) {
+                    g.mov(1, temp, src1);
+                    return temp;
+                } else {
                     return *reinterpret_cast<ngen::RegData *>(&src1);
-                g.mov(1, temp, src1);
-                return temp;
+                }
```

Comment:
I think we should add emulation specific to this case to the `s1Q` case below. It should be possible to accomplish this in 4 instructions (mul, mul, add, mov) rather than the proposed 7 (mov, mul, macl, mul, mach, add, mov) and it will also enable the `qwq` case as well. 

Longer term, it may also make sense to refactor so that `dstQ && s0Q` case contain the actual implementations, and the `dstQ && s1Q` contain the recursive implementation. I think this will provide some simplification as we know `S1` is not an immediate in that case, removing the above `mov` logic.

Reply:
We may be able to reduce it to 3 instructions, actually, using 48-bit precision accumulation in dw x w mul instructions. I created [MFDNN-103006](https://jira.devtools.intel.com/browse/MFDNN-13006) to track this optimization.

This PR fixes the broken layers, and I will make the suggested optimization changes in an upcoming PR since there's a bit of exploring to be done finding the best instructions to use.